src/ert/analysis/_es_update.py

```diff
@@ -43,6 +46,99 @@ class SmootherSnapshot:
     update_step_snapshots: Dict[str, "UpdateSnapshot"] = field(default_factory=dict)
 
 
+def _get_A_matrix(
+    temporary_storage: Dict[str, "npt.NDArray[Any]"],
+    parameters: List[update.Parameter],
+) -> Any:
```

Comment:
Could we be more specific than `Any`?
Same when defining `matrices`.

Reply:
numpy is strange when it comes to types. mypy complains that stack does not take a list of NDArrays..